sympy/solvers/tests/test_inequalities.py

```diff
@@ -291,6 +286,27 @@ def test_solve_univariate_inequality():
     assert isolve((x - 1)/den <= 0, x) == \
         Or(And(-oo < x, x < 1), And(S(1) < x, x < 2))
 
+    n = Dummy('n')
+    raises(NotImplementedError, lambda: isolve(Abs(x) <= n, x, relational=False))
+
+
+def test_trig_inequalities():
+    # all the inequalities are solved in a periodic interval.
+    assert isolve(sin(x) < S.Half, x, relational=False) == \
+        Union(Interval(0, pi/6, False, True), Interval(5*pi/6, 2*pi, True, True))
+    assert isolve(sin(x) > S.Half, x, relational=False) == \
+        Interval(pi/6, 5*pi/6, True, True)
+    assert isolve(cos(x) < S.Zero, x, relational=False) == \
+        Interval(pi/2, 3*pi/2, True, True)
+    assert isolve(cos(x) >= S.Zero, x, relational=False) == \
+        Union(Interval(0, pi/2), Interval(3*pi/2, 2*pi, False, True))
+
+    assert isolve(tan(x) < S.One, x, relational=False) == \
+        Union(Interval(0, pi/4, False, True), Interval(pi/2, pi, True, True))
+
+    assert isolve(sin(x) <= S.Zero, x, relational=False) == \
+        Union(FiniteSet(S(0)), Interval(pi, 2*pi, False, True))
```

Comment:
The inequalities are solved in `[0, 2*pi)`
Since `f(x + p) = f(x)`, it didn't make sense to me to include same repetitive values.
Hence, I have used an open interval.